homeassistant/components/zwave_js/sensor.py

```diff
@@ -467,6 +475,7 @@ def __init__(
         self._attr_device_info = {
             "identifiers": {get_device_id(self.client, self.node)},
         }
+        self._attr_icon = STATUS[node.status.name.lower()]
```

Comment:
I think you'll want to add this also under the `_status_changed` function as well so when the state changes the icon updates as well.

Reply:
A general question (maybe for @raman325?) , is there a reason entity attributes are preferred over property functions when the attribute is dynamic? Is it because it's assumed the node state changing is a "rare" event? As an entity attribute, you need to take care to update the attribute when the node status changes. With a property function there's no reason to, it always accesses the current value. With the former you do need to do as @firstof9 has suggested and update the attribute when the status changes.

Besides that, I would use `NodeStatus` as the key instead of strings (with typing). This avoids a conversion from the enum to a string and the mutation (`lower()`).

```python
from zwave_js_server.const import NodeStatus

STATUS_ICON_MAP: dict[NodeStatus, str] = {
    NodeStatus.ALIVE: "mdi:heart-pulse",
    NodeStatus.ASLEEP: "mdi:sleep",
    NodeStatus.AWAKE: "mdi:eye",
    NodeStatus.DEAD: "mdi:robot-dead",
}
DEFAULT_STATUS_ICON = "mdi:help-rhombus"


@dataclass
class ZwaveSensorEntityDescription(SensorEntityDescription):
    ...

    @property
    def icon(self) -> str | None:
        """Icon of the entity."""
        return STATUS_ICON_MAP.get(node.status, DEFAULT_STATUS_ICON)

    # Or set self._attr_icon on status changes

    @callback
    def _status_changed(self, _: dict) -> None:
        """Call when status event is received."""
        self._attr_native_value = self.node.status.name.lower()
        self._attr_icon = STATUS_ICON_MAP.get(node.status, DEFAULT_STATUS_ICON)
        self.async_write_ha_state()
```

Reply:
I think the goal is to not have to re-invent the wheel since the `Entity` class already has the property and is looking for `self._attr_icon`

Reply:
ill await Ramans view on the suggestions you both raised, for which I say thanks!

Reply:
Maybe, but I ask because the [documentation](https://developers.home-assistant.io/docs/core/entity#property-implementation) does not explain why you would choose one technique over the other and says both are valid. The examples both show when the attribute is static, in this case it's dynamic. It would seem less error prone to me (as evidenced by this PR) to use a property function for dynamic values, but there must be good reasons not to. :)

Reply:
I don't think one or the other is wrong, it's a matter of clarity and simplicity. Generally speaking using the entity attributes is good when the value it has is static or there's a single place where it gets updated, like `self._attr_native_value` in the example above. I would probably just use it under the property myself and get rid of `self._attr_icon` references since you are putting it in two places versus the one

Reply:
icon (and many other attributes) are only updated in 1 place (usually in init or update methods), so 1 line vs 4 saves code.
In this case, a property would be the best approach

Reply:
> Besides that, I would use NodeStatus as the key instead of strings (with typing). This avoids a conversion from the enum to a string and the mutation (lower()).

I agree with this

Reply:
thanks, Ill go ahead and revert what I did, and replace it with @kpine's suggestion above.